rfcs/80-curve-trait.md

```diff
@@ -0,0 +1,532 @@
+Feature Name: `curve-trait`
+
+## Summary
+
+This RFC introduces a general trait API, `Curve<T>`, for shared functionality of curves within Bevy's
+ecosystem. This encompasses both *abstract* curves, such as those produced by splines (and hence defined
+by equations), as well as *concrete* curves that are defined by sampling and interpolation (e.g. 
+animation keyframes), among others. 
+
+The purpose is to provide a common baseline of useful functionality that unites these disparate
+implementations, making curves of many kinds pleasant to work with for authors of plugins and games.
+
+## Motivation
```

Comment:
I'd like to see a list of concrete list of game-related use cases of curves as part of this section. It's helpful background, and justifies why curves are in bevy_math at all.